src/workflow/closure.rs

```diff
@@ -37,24 +37,25 @@ pub struct Closure {
     pub inputs: Input,
 }
 
-impl Into<Link<Closure>> for Closure {
-    fn into(self) -> Link<Closure> {
-        let mut closure_bytes = Vec::new();
-        <Closure as Into<Ipld>>::into(self).encode(DagCborCodec, &mut closure_bytes);
+impl TryFrom<Closure> for Link<Closure> {
+    type Error = anyhow::Error;
 
-        Link::new(Cid::new_v1(
+    fn try_from(closure: Closure) -> Result<Link<Closure>, Self::Error> {
+        let mut closure_bytes = Vec::new();
+        <Closure as Into<Ipld>>::into(closure).encode(DagCborCodec, &mut closure_bytes)?;
```

Comment:
I hate that the compiler needs me to write out `<Closure as Into<Ipld>>::into(closure)` with the burning passion of a thousand suns. Is there a better way?